src/sparsebase/utils/converter/converter.h

```diff
@@ -43,124 +45,279 @@ namespace converter {
 
 
 
-//! A conversion schema is a way to store a conversion to a particular type and
-//! context
-typedef std::vector<std::tuple<bool, std::type_index, context::Context *>>
-    ConversionSchemaConditional;
 
 //! Any function that matches the signature of this definition can be registered
 //! in Converter instances
-using ConditionalConversionFunction =
-    std::function<format::Format *(format::Format *, context::Context *)>;
+
+typedef std::function<format::Format *(format::Format *, context::Context *)>
+    ConversionFunction;
+
 
 //! A function type that returns true if the relevant conversion function can
 //! convert between the contexts in the parameters
-using EdgeConditional =
-    std::function<bool(context::Context *, context::Context *)>;
+typedef
+std::function<bool(context::Context *, context::Context *)>
+    ConversionCondition;
+
+//! A single conversion step composed of a conversion function and a context to use for the function
+typedef std::tuple<ConversionFunction, context::Context *> ConversionStep;
```

Comment:
Is there a particular reason this is not an `std::pair`. Is this where we would add weights?

Reply:
No, there isn't really a reason. It can be a pair. I opted to associate the cost with an entire chain rather than a step, but now that I think about it it's more general to associate steps with costs. I'll add that.

Reply:
That cost is currently not being used, but in the future, it might become useful.